app/models/namespace.rb

```diff
@@ -40,13 +40,22 @@ class Namespace < ActiveRecord::Base
   belongs_to :registry
   belongs_to :team
 
-  validates :public, inclusion: { in: [true] }, if: :global?
+  enum visibility: [:visibility_private, :visibility_protected, :visibility_public]
+
+  validate :global_namespace_cannot_be_private
```

Comment:
Unfortunately, the following code (and some variations) didn't work:
```
validates :visibility, inclusion: { in: [:visibility_public, :visibility_private] }, if: :global?
```
Using a separate function works perfectly though.